BraintreeCore/src/main/java/com/braintreepayments/api/PostalAddressParser.kt

```diff
@@ -0,0 +1,89 @@
+package com.braintreepayments.api
+
+import androidx.annotation.RestrictTo
+import org.json.JSONObject
+
+@RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+object PostalAddressParser {
+    const val RECIPIENT_NAME_KEY = "recipientName"
+    const val STREET_ADDRESS_KEY = "street1"
+    const val EXTENDED_ADDRESS_KEY = "street2"
+    const val LOCALITY_KEY = "city"
+    const val COUNTRY_CODE_ALPHA_2_KEY = "country"
+    const val POSTAL_CODE_KEY = "postalCode"
+    const val REGION_KEY = "state"
+    const val LINE_1_KEY = "line1"
+    const val LINE_2_KEY = "line2"
+    const val COUNTRY_CODE_KEY = "countryCode"
+    const val USER_ADDRESS_NAME_KEY = "name"
+    const val USER_ADDRESS_PHONE_NUMBER_KEY = "phoneNumber"
+    const val USER_ADDRESS_ADDRESS_1_KEY = "address1"
+    const val USER_ADDRESS_ADDRESS_2_KEY = "address2"
+    const val USER_ADDRESS_ADDRESS_3_KEY = "address3"
+    const val USER_ADDRESS_ADDRESS_4_KEY = "address4"
+    const val USER_ADDRESS_ADDRESS_5_KEY = "address5"
+    const val USER_ADDRESS_POSTAL_CODE_KEY = "postalCode"
+    const val USER_ADDRESS_SORTING_CODE_KEY = "sortingCode"
+    const val USER_ADDRESS_COUNTRY_CODE_KEY = "countryCode"
+    const val USER_ADDRESS_LOCALITY_KEY = "locality"
+    const val USER_ADDRESS_ADMINISTRATIVE_AREA_KEY = "administrativeArea"
+    const val COUNTRY_CODE_UNDERSCORE_KEY = "country_code"
+    const val POSTAL_CODE_UNDERSCORE_KEY = "postal_code"
+    const val RECIPIENT_NAME_UNDERSCORE_KEY = "recipient_name"
+
+    @JvmStatic
+    fun fromJson(accountAddress: JSONObject?): PostalAddress =
+        // If we don't have an account address, return an empty PostalAddress.
+        accountAddress?.let {
+            var streetAddress = Json.optString(accountAddress, STREET_ADDRESS_KEY, null)
+            var extendedAddress = Json.optString(accountAddress, EXTENDED_ADDRESS_KEY, null)
+            var countryCodeAlpha2 = Json.optString(accountAddress, COUNTRY_CODE_ALPHA_2_KEY, null)
+
+            //Check alternate keys
+            streetAddress = streetAddress ?: Json.optString(accountAddress, LINE_1_KEY, null)
+            extendedAddress = extendedAddress ?: Json.optString(accountAddress, LINE_2_KEY, null)
+            countryCodeAlpha2 = countryCodeAlpha2 ?: Json.optString(accountAddress, COUNTRY_CODE_KEY, null)
+
+            // If this is a UserAddress-like JSON, parse it as such
+            if (streetAddress == null && Json.optString(
+                    accountAddress,
+                    USER_ADDRESS_NAME_KEY,
+                    null
+                ) != null
+            ) {
+                return@let fromUserAddressJson(accountAddress)
+            }
+
+            PostalAddress().apply {
```

Comment:
yes. `apply()` is commonly used for object configuration. Without `apply` you would have to first create an empty `PostalAddress`, assign to a variable, then set all the variables and later return it. Here we use less code, because `apply` returns the object.
[Here are the official docs](https://kotlinlang.org/docs/scope-functions.html#apply), for more clarification